### ISSUE TITLE
':' character in a header unit or value causes incorrect output from parser

### ISSUE THREAD

**Issue by kinverarity1** [kinverarity1/lasio#201]
Lines from a LAS header

```
TCS .hh:mm 21:30 23-JAN-2001:   Time Circ. Stopped
TIML.hh:mm 23:15 23-JAN-2001:   Time Logger At Bottom
```

Current behaviour - it breaks the unit/descr at the first colon from the left. 

```
HeaderItem(mnemonic=TCS, unit=hh, value=, descr=mm 21:30 23-JAN-2001:   Time Circ. Stopped, original_mnemonic=TCS)
HeaderItem(mnemonic=TIML, unit=hh, value=, descr=mm 23:15 23-JAN-2001:   Time Logger At Bottom, original_mnemonic=TIML)
``` 

Expected behavior:

```
HeaderItem(mnemonic=TCS, unit=hh:mm, value=21:30 23-JAN-2001, descr=Time Circ. Stopped, original_mnemonic=TCS)
HeaderItem(mnemonic=TIML, unit=hh:mm, value=23:15 23-JAN-2001, descr=Time Logger At Bottom, original_mnemonic=TIML)
```

**Comment by kinverarity1** [comment]
I could achieve the expected behaviour in this example by breaking at the first colon reading from the right. That would introduce a new bug in the case that the descr has a colon in it, but I feel that would be less intrusive. The current behaviour breaks the unit and value (both important), and descr (not so important). The proposed solution would break only the value (important) and descr (not so important), but read the unit properly.

The ideal solution is more complex: exploit the fact that the value/descr colon is usually the same column for every line in the section, and use a likelihood function rather than a regexp to detect the value/descr colon. I don't have time to implement this at the moment.

**Comment by RomCoch** [comment]
I has this problem too, specifically with with "x:long_value y:lat_value" as loc value.


> 
> 
> I could achieve the expected behaviour in this example by breaking at the first colon reading from the right. That would introduce a new bug in the case that the descr has a colon in it, but I feel that would be less intrusive. The current behaviour breaks the unit and value (both important), and descr (not so important). The proposed solution would break only the value (important) and descr (not so important), but read the unit properly.
> 
> The ideal solution is more complex: exploit the fact that the value/descr colon is usually the same column for every line in the section, and use a likelihood function rather than a regexp to detect the value/descr colon. I don't have time to implement this at the moment.

Your first idea is consistent with LAS2.0 specs, which specify that line delimiters between mnemonic, units, value and description are "a) the first dot in a line, b) the first space after the first dot in a line c) the last colon in a line". And that correction is far more easy to proceed (minor modification of a regexp in reader.py: r"(?P<value>[^:]*):" --> r"(?P<value>.*):").

According to LAS2.0 specs, the descr should't hence include colons, so there would be no side effect in reading LAS2.0 files. I don't know about other versions, but it would at worst only break the descr, which is, I agree with you, really less damageable than breaking the units or the value.

**Comment by dcslagel** [comment]
@RomCoch , @kinverarity1 

Pull request https://github.com/kinverarity1/lasio/pull/298  to address this issue incorporates changes from your work.

It incorporates test .las file examples from
https://github.com/kinverarity1/lasio/tree/issue201-multiple-colons

And code solutions from :
RomCoch@71e97fd
and
RomCoch@156def9

The test cases can be run with

pytest -s -v tests/test_read.py::test_issue_201_non_delimiter_colon_start
pytest -s -v tests/test_read.py::test_issue_201_non_delimiter_colon_end

The test suite passed with this change included.

DC

**Comment by dcslagel** [comment]
@kinverarity1,

Can this issue be closed now that pull request #298 is merged or is there additional work that needs to be done?

Thank you,
DC

**Comment by kinverarity1** [comment]
Yes, certainly - thanks for the PR 👍
